src/Scanner/ChainScanner.ts

```diff
@@ -32,48 +32,71 @@ export default class ChainScanner extends CacheScanner {
+      this.stopWatcher(this.buckets.currentBuckets.blockBucket);
+
+      // If we are only doing one bucket step up we only need to start one watcher.
+      this.buckets.currentBuckets.blockBucket = buckets.currentBuckets.blockBucket;
+      this.buckets.nextBuckets.blockBucket = await this.watchRequestsByBucket(
+        buckets.nextBuckets.blockBucket,
+        this.buckets.nextBuckets.blockBucket
+      );
+    } else {
+      // Start watching the current buckets right away.
+      this.buckets.currentBuckets.blockBucket = await this.watchRequestsByBucket(
+        buckets.currentBuckets.blockBucket,
+        this.buckets.currentBuckets.blockBucket
+      );
+      this.buckets.nextBuckets.blockBucket = await this.watchRequestsByBucket(
+        buckets.nextBuckets.blockBucket,
+        this.buckets.nextBuckets.blockBucket
+      );
+    }
+
+    if (this.buckets.nextBuckets.timestampBucket === buckets.currentBuckets.timestampBucket) {
+      this.stopWatcher(this.buckets.currentBuckets.timestampBucket);
+
+      this.buckets.currentBuckets.timestampBucket = buckets.currentBuckets.timestampBucket;
+      this.buckets.nextBuckets.timestampBucket = await this.watchRequestsByBucket(
+        buckets.nextBuckets.timestampBucket,
+        this.buckets.nextBuckets.timestampBucket
+      );
+    } else {
+      this.buckets.currentBuckets.timestampBucket = await this.watchRequestsByBucket(
+        buckets.currentBuckets.timestampBucket,
+        this.buckets.currentBuckets.timestampBucket
+      );
+      this.buckets.nextBuckets.timestampBucket = await this.watchRequestsByBucket(
+        buckets.nextBuckets.timestampBucket,
+        this.buckets.nextBuckets.timestampBucket
+      );
+    }
 
     this.config.logger.debug(`Buckets: after current buckets=${JSON.stringify(this.buckets)}`);
   }
 
-  public async watchRequestsByBucket(bucket: Bucket, previousBucket: Bucket): Promise<number> {
-    const reqFactory = await this.requestFactory;
-    const handleRequest = this.handleRequest.bind(this);
-    let currentBucket = previousBucket;
-
+  public async watchRequestsByBucket(bucket: Bucket, previousBucket: Bucket): Promise<Bucket> {
     if (bucket !== previousBucket) {
       await this.stopWatcher(previousBucket);
+      return this.startWatcher(bucket);
+    }
 
-      try {
-        const watcher = await reqFactory.watchRequestsByBucket(bucket, handleRequest);
-        this.eventWatchers[bucket] = watcher;
-        currentBucket = bucket;
+    return previousBucket;
+  }
 
-        this.config.logger.debug(`Buckets: Watcher for bucket=${bucket} has been started`);
-      } catch (err) {
-        this.config.logger.error(`Buckets: Starting bucket=${bucket} watching failed!`);
-      }
+  protected async startWatcher(bucket: Bucket): Promise<Bucket> {
+    const reqFactory = await this.requestFactory;
+    const handleRequest = this.handleRequest.bind(this);
```

Comment:
Can we move this line to constructor, so we don't override this function on each `startWatcher` call?